functions/pipeline.py

```diff
@@ -0,0 +1,467 @@
+
+    raw_input_data_file: str
+        File containing raw expression data
+    '''
+
+    # Read in config variables
+    params = utils.read_config(config_file)
+
+    # Load parameters
+    dataset_name = params['dataset_name']
+
+    # Read data
+    rpkm_data = pd.read_table(
+        raw_input_data_file,
+        header=0,
+        sep='\t',
+        index_col=0)
+
+    print(rpkm_data.shape)
+
+    # 0-1 normalize per gene
+    rnaseq_scaled_df = preprocessing.MinMaxScaler().fit_transform(rpkm_data)
+    rnaseq_scaled_df = pd.DataFrame(rnaseq_scaled_df,
+                                    columns=rpkm_data.columns,
+                                    index=rpkm_data.index).T
+
+    # Save scaled data
+    normalized_data_file = os.path.join(
+        base_dir,
+        dataset_name,
+        "data",
+        "input",
+        "recount2_gene_normalized_data.tsv.xz")
+
+    rnaseq_scaled_df.to_csv(normalized_data_file,
+                            sep='\t', compression='xz')
+
+    del rnaseq_scaled_df
+    gc.collect()
+
+
+def create_experiment_id_file(metadata_file,
+                              input_data_file,
+                              output_file,
+                              config_file):
+    '''
+    Create file with experiment ids that are associated with expression data
+
+    Arguments
+    ----------
+    metadata_file: str
+        File containing metadata annotations per sample
+
+    input_data_file: str
+        File containing normalized expression data
+
+    output_file: str
+        File containing experiment ids with expression data and sample annotations
+
+    config_file: str
+        File containing user defined parameters
+
+    '''
+    # Read in metadata
+    metadata = pd.read_table(
+        metadata_file,
+        header=0,
+        sep='\t',
+        index_col=0)
+
+    # Read in expression data
+    normalized_data = pd.read_table(
+        input_data_file,
+        header=0,
+        sep='\t',
+        index_col=0).T
+
+    # Read in config variables
+    params = utils.read_config(config_file)
+
+    # Load parameters
+    sample_id_colname = params['metadata_colname']
+    dataset_name = params['dataset_name']
+
+    # Get sample id that maps between metadata and expression files
+    map_experiment_sample = metadata[[sample_id_colname]]
+
+    # Get all unique experiment ids
+    experiment_ids = np.unique(np.array(map_experiment_sample.index)).tolist()
+    print("There are {} experiments in the compendium".format(
+        len(experiment_ids)))
+
+    # Get only sample ids that have expression data available
+    sample_ids_with_gene_expression = list(normalized_data.index)
+
+    # Get associated experiment ids with expression data
+    experiment_ids_with_gene_expression = []
+
+    for experiment_id in experiment_ids:
+
+        if "Human" in dataset_name:
+            # Some project id values are descriptions
+            # We will skip these
+            if len(experiment_id) == 9:
```

Comment:
Where does the number 9 come from here? Are you sure that it will be true for all experiment_ids?

Reply:
This was manually coded by me based looking at the metadata sheet.  This will not necessarily be true for all incoming data.  I should add a comment about this in the readme as something else for the user to change.